lib/std/debug.zig

```diff
@@ -2307,16 +2307,284 @@ fn handleSegfaultLinux(sig: i32, info: *const os.siginfo_t, ctx_ptr: *const c_vo
 }
 
 fn handleSegfaultWindows(info: *windows.EXCEPTION_POINTERS) callconv(.Stdcall) c_long {
-    const exception_address = @ptrToInt(info.ExceptionRecord.ExceptionAddress);
     switch (info.ExceptionRecord.ExceptionCode) {
-        windows.EXCEPTION_DATATYPE_MISALIGNMENT => panicExtra(null, exception_address, "Unaligned Memory Access", .{}),
-        windows.EXCEPTION_ACCESS_VIOLATION => panicExtra(null, exception_address, "Segmentation fault at address 0x{x}", .{info.ExceptionRecord.ExceptionInformation[1]}),
-        windows.EXCEPTION_ILLEGAL_INSTRUCTION => panicExtra(null, exception_address, "Illegal Instruction", .{}),
-        windows.EXCEPTION_STACK_OVERFLOW => panicExtra(null, exception_address, "Stack Overflow", .{}),
+        windows.EXCEPTION_DATATYPE_MISALIGNMENT => handleSegfaultWindowsExtra(info, 0, "Unaligned Memory Access"),
+        windows.EXCEPTION_ACCESS_VIOLATION => handleSegfaultWindowsExtra(info, 1, null),
+        windows.EXCEPTION_ILLEGAL_INSTRUCTION => handleSegfaultWindowsExtra(info, 2, null),
+        windows.EXCEPTION_STACK_OVERFLOW => handleSegfaultWindowsExtra(info, 0, "Stack Overflow"),
         else => return windows.EXCEPTION_CONTINUE_SEARCH,
     }
 }
 
+// zig won't let me use an anon enum here
```

Comment:
Why is that? What error does it give?

Reply:
```
D:\zig>zig run ill.zig -target x86_64-windows-gnu
.\lib\zig\std\debug.zig:2311:85: error: expected type 'std.debug.enum:2320:80', found 'std.debug.enum:2320:80'
        windows.EXCEPTION_DATATYPE_MISALIGNMENT => handleSegfaultWindowsExtra(info, .fmt, "Unaligned Memory Access"),
                                                                                    ^
.\lib\zig\std\debug.zig:2320:80: note: std.debug.enum:2320:80 declared here
fn handleSegfaultWindowsExtra(info: *windows.EXCEPTION_POINTERS, comptime msg: enum {fmt, segv, ill}, comptime format: ?[]const u8) noreturn {
```

Reply:
Ah the infamous `enum` bug, I think it stops being a problem if you don't make it a comptime param.

Reply:
Nope, comptime does not make a difference

Reply:
Is there an issue for this? I'll take a look

Reply:
#3707, the anonymous enum is analyzed twice and produces two disjoint types.